modules/virt-accessing-vm-yaml-ssh.adoc

```diff
@@ -0,0 +1,159 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/virt-accessing-vm-consoles.adoc
+
+[id="virt-accessing-vm-yaml-ssh_{context}"]
+= Accessing a virtual machine via SSH with YAML configurations
+
+You can enable a SSH connection to a virtual machine (VM) without the need to run the `virtctl expose` command. When the YAML file for the VM and the YAML file for the service are configured and applied, the service forwards the SSH traffic to the VM.
+
+The following examples show the configurations for the VM's YAML file and the service YAML file.
+
+.Prerequisites
+* Install the OpenShift CLI (`oc`).
+* Create a namespace for the VM's YAML file by using the `oc create namespace` command and specifying a name for the namespace.
+
+.Procedure
+. In the YAML file for the VM, add the label and a value for exposing the service for SSH connections. Enable the `masquerade` feature for the interface.
+
++
+.Example output
+[source,yaml]
+----
+...
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  namespace: ssh-ns <1>
+  name: vm-ssh
+spec:
+  running: false
+  template:
+   metadata:
+    labels:
+     kubevirt.io/vm: vm-ssh <2>
+  spec:
+   domain:
+    devices:
+     disks:
+      - disk:
+          bus: virtio
+        name: containerdisk
+      - disk:
+          bus: virtio
+        name: cloudinitdisk
+      interfaces:
+      - masquerade: {} <3>
+        name: testmasquerade <4>
+      rng: {}
+    machine:
+      type: ""
+    resources:
+      requests:
+        memory: 1024M
+  networks:
+  - name: testmasquerade
+    pod: {}
+  volumes:
+  - name: containerdisk
+    containerDisk:
+      image: kubevirt/fedora-cloud-container-disk-demo
+  - name: cloudinitdisk
+    cloudInitNocloud:
+      userData: |
+        #!/bin/bash
+        echo "fedora" | passwd fedora --stdin
+...
+----
+<1> Name of the namespace created by the `oc create namespace` command.
+<2> This label is used by the service to identify the virtual machine instances that are enabled for SSH traffic connections. The label can be any `key:value` pair that is added as a `label` to this YAML file and as a `selector` in the service YAML file.
+<3> The interface type is `masquerade`.
+<4> The name of this interface is `testmasquerade`.
+
+. Create the VM:
++
+[source,terminal]
+----
+oc create -f <path_for_the_VM_YAML_file>
+----
++
+
+. Start the VM:
++
+[source,terminal]
+----
+virtctl start vm-ssh
+----
++
+
+. In the YAML file for the service, specify the service name, port number, and the target port.
+
++
+.Example output
+[source,yaml]
+----
+...
+apiVersion: v1
+kind: Service
+metadata:
+  name: svc-ssh <1>
+  namespace: ssh-ns <2>
+spec:
+  ports:
+   - targetPort: 22 <3>
+     protocol: TCP
+     port: 27017
+  selector:
+     special: vm-ssh <4>
+  type: NodePort
+...
+----
+<1> Name of the namespace created by the `oc create namespace` command.
```

Comment:
You confused <1> and <2>. In the yaml - you placed <1> next to `name` and <2> next to `namespace`.

Reply:
Updated content.